modules/rules_foreign_cc/0.7.1/patches/label.patch

```diff
@@ -0,0 +1,159 @@
++        toolchain_name = "rules_foreign_cc_framework_toolchain_" + item.file.split(":")[1][:-len(".bzl")]
+ 
+         framework_toolchain_repository(
+             name = toolchain_name,
+diff --git a/foreign_cc/private/framework/toolchains/access.bzl b/foreign_cc/private/framework/toolchains/access.bzl
+index 16d05d1..3fb5988 100644
+--- a/foreign_cc/private/framework/toolchains/access.bzl
++++ b/foreign_cc/private/framework/toolchains/access.bzl
+@@ -16,7 +16,7 @@ def create_context(ctx):
+             - prelude (dict): A cache for rendered functions
+     """
+     return struct(
+-        shell = ctx.toolchains[str(Label("//foreign_cc/private/framework:shell_toolchain"))].commands,
++        shell = ctx.toolchains["@rules_foreign_cc//foreign_cc/private/framework:shell_toolchain"].commands,
+         prelude = {},
+     )
+ 
+diff --git a/foreign_cc/private/framework/toolchains/mappings.bzl b/foreign_cc/private/framework/toolchains/mappings.bzl
+index ff7849b..4a9a75d 100644
+--- a/foreign_cc/private/framework/toolchains/mappings.bzl
++++ b/foreign_cc/private/framework/toolchains/mappings.bzl
+@@ -23,24 +23,24 @@ TOOLCHAIN_MAPPINGS = [
+         exec_compatible_with = [
+             "@platforms//os:linux",
+         ],
+-        file = Label("@rules_foreign_cc//foreign_cc/private/framework/toolchains:linux_commands.bzl"),
++        file = "@rules_foreign_cc//foreign_cc/private/framework/toolchains:linux_commands.bzl",
+     ),
+     _toolchain_mapping(
+         exec_compatible_with = [
+             "@platforms//os:freebsd",
+         ],
+-        file = Label("@rules_foreign_cc//foreign_cc/private/framework/toolchains:freebsd_commands.bzl"),
++        file = "@rules_foreign_cc//foreign_cc/private/framework/toolchains:freebsd_commands.bzl",
+     ),
+     _toolchain_mapping(
+         exec_compatible_with = [
+             "@platforms//os:windows",
+         ],
+-        file = Label("@rules_foreign_cc//foreign_cc/private/framework/toolchains:windows_commands.bzl"),
++        file = "@rules_foreign_cc//foreign_cc/private/framework/toolchains:windows_commands.bzl",
+     ),
+     _toolchain_mapping(
+         exec_compatible_with = [
+             "@platforms//os:macos",
+         ],
+-        file = Label("@rules_foreign_cc//foreign_cc/private/framework/toolchains:macos_commands.bzl"),
++        file = "@rules_foreign_cc//foreign_cc/private/framework/toolchains:macos_commands.bzl",
+     ),
+ ]
+diff --git a/toolchains/native_tools/tool_access.bzl b/toolchains/native_tools/tool_access.bzl
+index 5dc5404..cc07770 100644
+--- a/toolchains/native_tools/tool_access.bzl
++++ b/toolchains/native_tools/tool_access.bzl
+@@ -24,25 +24,25 @@ def access_tool(toolchain_type_, ctx, tool_name):
+     )
+ 
+ def get_autoconf_data(ctx):
+-    return _access_and_expect_label_copied(str(Label("//toolchains:autoconf_toolchain")), ctx, "autoconf")
++    return _access_and_expect_label_copied("//toolchains:autoconf_toolchain", ctx, "autoconf")
```

Comment:
also to confirm, these will definitely work, since they're used as labels to identify toolchain infos and are resolved in the context of the this .bzl file.